### DISCUSSION TITLE
bug: YouTube RAG fails when subtitles are not available

### DISCUSSION THREAD

**Discussion by bannert1337** [open-webui/open-webui#3446] [General]
# Bug Report

## Description

**Bug Summary:**
When referencing YouTube videos with subtitles not available, an error occurs, and the YouTube video cannot be added as a reference.

**Steps to Reproduce:**
1. Attempt to reference a YouTube video without available subtitles using the YouTube RAG.
2. Observe the error message: "Something went wrong :/ Could not retrieve a transcript for the video."

**Expected Behavior:**
The system should gracefully handle videos without available subtitles and provide a more informative error message or alternative behavior.

**Actual Behavior:**
The system fails and outputs an error message, preventing the YouTube video from being referenced.

## Environment

- **Open WebUI Version:** [Please specify]
- **Ollama (if applicable):** [Please specify]

- **Operating System:** [Please specify]
- **Browser (if applicable):** [Please specify]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

The error message received is as follows:

```
Something went wrong :/
Could not retrieve a transcript for the video https://www.youtube.com/watch?v=6nuPVOKfq18!
This is most likely caused by: No transcripts were found for any of the requested language codes: ['en']
For this video (6nuPVOKfq18) transcripts are available in the following languages:
(MANUALLY CREATED) None 
(GENERATED) - de ("German (auto-generated)")
[TRANSLATABLE] (TRANSLATION LANGUAGES) - af ("Afrikaans") - ak ("Akan") - sq ("Albanian") - am ("Amharic") - ar ("Arabic") - hy ("Armenian") - as ("Assamese") - ay ("Aymara") - az ("Azerbaijani") - bn ("Bangla") - eu ("Basque") - be ("Belarusian") - bho ("Bhojpuri") - bs ("Bosnian") - bg ("Bulgarian") - my ("Burmese") - ca ("Catalan") - ceb ("Cebuano") - zh-Hans ("Chinese (Simplified)") - zh-Hant ("Chinese (Traditional)") - co ("Corsican") - hr ("Croatian") - cs ("Czech") - da ("Danish") - dv ("Divehi") - nl ("Dutch") - en ("English") - eo ("Esperanto") - et ("Estonian") - ee ("Ewe") - fil ("Filipino") - fi ("Finnish") - fr ("French") - gl ("Galician") - lg ("Ganda") - ka ("Georgian") - de ("German") - el ("Greek") - gn ("Guarani") - gu ("Gujarati") - ht ("Haitian Creole") - ha ("Hausa") - haw ("Hawaiian") - iw ("Hebrew") - hi ("Hindi") - hmn ("Hmong") - hu ("Hungarian") - is ("Icelandic") - ig ("Igbo") - id ("Indonesian") - ga ("Irish") - it ("Italian") - ja ("Japanese") - jv ("Javanese") - kn ("Kannada") - kk ("Kazakh") - km ("Khmer") - rw ("Kinyarwanda") - ko ("Korean") - kri ("Krio") - ku ("Kurdish") - ky ("Kyrgyz") - lo ("Lao") - la ("Latin") - lv ("Latvian") - ln ("Lingala") - lt ("Lithuanian") - lb ("Luxembourgish") - mk ("Macedonian") - mg ("Malagasy") - ms ("Malay") - ml ("Malayalam") - mt ("Maltese") - mi ("Māori") - mr ("Marathi") - mn ("Mongolian") - ne ("Nepali") - nso ("Northern Sotho") - no ("Norwegian") - ny ("Nyanja") - or ("Odia") - om ("Oromo") - ps ("Pashto") - fa ("Persian") - pl ("Polish") - pt ("Portuguese") - pa ("Punjabi") - qu ("Quechua") - ro ("Romanian") - ru ("Russian") - sm ("Samoan") - sa ("Sanskrit") - gd ("Scottish Gaelic") - sr ("Serbian") - sn ("Shona") - sd ("Sindhi") - si ("Sinhala") - sk ("Slovak") - sl ("Slovenian") - so ("Somali") - st ("Southern Sotho") - es ("Spanish") - su ("Sundanese") - sw ("Swahili") - sv ("Swedish") - tg ("Tajik") - ta ("Tamil") - tt ("Tatar") - te ("Telugu") - th ("Thai") - ti ("Tigrinya") - ts ("Tsonga") - tr ("Turkish") - tk ("Turkmen") - uk ("Ukrainian") - ur ("Urdu") - ug ("Uyghur") - uz ("Uzbek") - vi ("Vietnamese") - cy ("Welsh") - fy ("Western Frisian") - xh ("Xhosa") - yi ("Yiddish") - yo ("Yoruba") - zu ("Zulu")
If you are sure that the described cause is not responsible for this error and that a transcript should be retrievable, please create an issue at https://github.com/jdepoix/youtube-transcript-api/issues.
Please add which version of youtube_transcript_api you are using and provide the information needed to replicate the error.
Also make sure that there are no open issues which already describe your problem!
```

This issue suggests the need for handling cases where transcripts are not available, ensuring better user experience.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by justinh-rahb** [comment]
Behaving as expected. YouTube RAG needs a transcript to work... Not sure how it's expected to resolve this.

**Reply by thiswillbeyourgithub** [reply]
I disagree, the audio is easy to get, we can then directly send the audio to a STT engine end use the output as text.

I made that for [my RAG project WDoc](https://github.com/thiswillbeyourgithub/WDoc/) and it works great. [Here's the relevant code](https://github.com/thiswillbeyourgithub/WDoc/blob/634f54d47c1104f727902e74e565af3a9fe4d725/WDoc/utils/loaders.py#L596) where it uses either openai or deepgram as an engine for the STT.

By the way, [I also made a python function that scrapes video on any website, that is useful if youtube_dl is down OR if you want to scrape a video on another website than youtube](https://github.com/thiswillbeyourgithub/WDoc/blob/634f54d47c1104f727902e74e565af3a9fe4d725/WDoc/utils/loaders.py#L2181)
